.circleci/config.yml

```diff
@@ -177,9 +229,76 @@ jobs:
           path: /tmp/resource-usage
           destination: resource-usage
 
+  unit-test-windows:
+    executor:
+      name: win/default
+      shell: bash.exe
+    steps:
+      - dependencies-windows
+
+      # - run:
+      #     name: Memory Monitor
+      #     command: |
+      #       mkdir /tmp/resource-usage
+      #       export FILE=/tmp/resource-usage/memory.txt
+      #       while true; do
+      #         ps -u root eo pid,%cpu,%mem,args,uname --sort=-%mem >> $FILE
+      #         echo "----------" >> $FILE
+      #         sleep 1
+      #       done
+      #     background: true
+
+      # Note: The setup.py checksum in the cache name ensures that we retrain our expert
+      # policies, which are cached in the pytest cache, whenever setup.py changes. This
+      # is just a rough heuristic to decide when the experts should be retrained but
+      # still better than nothing.
+      - restore_cache:
+          keys:
+            - v8-win-pytest-cache-{{ checksum "setup.py" }}
+            # This prefix-matched key restores the most recent pytest cache with any
+            # checksum. If that cached policy happens to be still loadable, we avoid
+            # retraining. See https://circleci.com/docs/2.0/caching/#restoring-cache
+            - v8-win-pytest-cache-
+          paths:
+            - .pytest_cache
+
+      - run:
+          name: run tests
+          command: |
+            $ErrorActionPreference = "Stop"  # exit immediately on any error
+            .\venv\Scripts\activate
+            # Xdummy-entrypoint.py: starts an X server and sets DISPLAY, then runs wrapped command.
+            pytest -n 4 --cov=venv\Lib\site-packages\imitation `
+            --cov=tests --junitxml=\tmp\test-reports\junit.xml -k 'not run_example_sh_scripts' `
+            -vv tests\ --ignore tests\test_experiments.py ` --ignore tests\test_scripts.py
+            if ($LASTEXITCODE -ne 0) { throw "pytest failed" }
```

Comment:
Do we need this as well as ``$ErrorActionPreference = "Stop"``?

Reply:
As far as I understand, `$ErrorActionPreference = "Stop"` works when the error is generated through a powershell command (cmdlet), but it does not work when the error is generated through exe files. In such cases, we have to manually check through the $LASTEXITCODE variable whether the command returned with an error or not. 
So we definitely require the if condition. However, I'm not sure how helpful is setting `$ErrorActionPreference = "Stop"`.

You can look up more about this issue in the accepted answer of [this](https://stackoverflow.com/questions/9948517/how-to-stop-a-powershell-script-on-the-first-error) StackOverflow question.

Reply:
OK, in that case definitely keep this line, and I'm fine having the ErrorActionPreference = "Stop" as well in case we have any errors in (future) cmdlet's too.